src/yardstick/store/labels.py

```diff
@@ -96,40 +106,55 @@ def save(label_entries: List[artifact.LabelEntry], store_root: Optional[str] = N
         with open(path, "w", encoding="utf-8") as data_file:
             if len(destination_label_entries) == 1:
                 json.dump(
-                    omitempty(list(destination_label_entries)[0].to_dict()),  # pylint: disable=not-callable
+                    omitempty(
+                        next(iter(destination_label_entries)).to_dict(),  # type: ignore[attr-defined]
```

Comment:
Why change `list(...)` to `next(...)`? Those seem to have very different semantics.

Reply:
this was a ruff autofix (and I confirmed its correct in functionality). This is based on https://docs.astral.sh/ruff/rules/unnecessary-iterable-allocation-for-first-element/ ... the semantics would be different if it wasn't for the `[0]` access at the end.